src/Missings.jl

```diff
@@ -208,6 +208,447 @@ missing
+    fill!(out, missing)
+    out[nonmissingmask] .= res
+    out
+end
+
+function maybespread_missing(
+    res::T,
+    spread::SpreadDefault,
+    vecs::Tuple,
+    nonmissinginds::AbstractVector{<:Integer},
+    nonmissingmask::AbstractVector{<:Bool})::T where{T}
+
+    res
+end
+
+function maybespread_missing(
+    res::AbstractVector{T},
+    spread::SpreadDefault,
+    vecs::Tuple,
+    nonmissinginds::AbstractVector{<:Integer},
+    nonmissingmask::AbstractVector{<:Bool})::AbstractVector{Union{Missing, T}} where {T}
+
+    spread_missing(res, vecs, nonmissinginds, nonmissingmask)
+end
+
+function maybespread_missing(
+    res::T,
+    spread::SpreadNonMissing,
+    vecs::Tuple,
+    nonmissinginds::AbstractVector{<:Integer},
+    nonmissingmask::AbstractVector{<:Bool})::AbstractVector{Union{Missing, T}} where{T}
+
+    out = Vector{Union{typeof(res), Missing}}(undef, length(vecs[1]))
+    fill!(out, missing)
+    out[nonmissinginds] .= Ref(res)
+    out
+end
+
+function maybespread_missing(
+    res::AbstractVector{T},
+    spread::SpreadNonMissing,
+    vecs::Tuple,
+    nonmissinginds::AbstractVector{<:Integer},
+    nonmissingmask::AbstractVector{<:Bool})::AbstractVector{Union{Missing, T}} where {T}
+
+    spread_missing(res, vecs, nonmissinginds, nonmissingmask)
+end
+
+function maybespread_missing(
+    res::T,
+    spread::SpreadNone,
+    vecs::Tuple,
+    nonmissinginds::AbstractVector{<:Integer},
+    nonmissingmask::AbstractVector{<:Bool})::T where {T}
+
+    res
+end
+
+function maybespread_missing(
+    res::T,
+    spread::SpreadAll,
+    vecs::Tuple,
+    nonmissinginds::AbstractVector{<:Integer},
+    nonmissingmask::AbstractVector{<:Bool})::AbstractVector{T} where {T}
+
+    out = Vector{typeof(res)}(undef, length(first(vecs)))
+    out .= Ref(res)
+    out
+end
+
+function maybespread_missing(
+    res::AbstractVector,
+    spread::SpreadAll,
+    vecs::Tuple,
+    nonmissinginds::AbstractVector{<:Integer},
+    nonmissingmask::AbstractVector{<:Bool})
+
+    throw(ArgumentError("spreadmissings with :all on vector output is reserved"))
+end
+
+function spread_nomissing(
+    res::AbstractVector{T},
+    vecs::Tuple)::typeof(res) where {T}
+
+    if length(res) != length(first(vecs))
+        s = "When spreading a vector result with `spread=$(S)`, " *
+            "length of output must match number of jointly non-"
+            "missing values in inputs "
+            "(got $(length(res)) and $(length(first(vecs)))).".
+        throw(DimensionMismatch(s))
+    end
+    res
+end
+
+function maybespread_nomissing(
+    res::T,
+    spread::SpreadDefault,
+    vecs::Tuple)::T where{T}
+
+    res
+end
+
+function maybespread_nomissing(
+    res::AbstractVector{T},
+    spread::SpreadDefault,
+    vecs::Tuple)::typeof(res) where {T}
+
+    spread_nomissing(res, vecs)
+end
+
+function maybespread_nomissing(
+    res::T,
+    spread::SpreadNonMissing,
+    vecs::Tuple)::Vector{T} where{T}
```

Comment:
Are all these return type annotations needed? In many cases they seem redundant.